src/app/researcher/study/request/[memberSlug]/study-proposal.tsx

```diff
@@ -5,19 +5,130 @@ import { Button, Group, Stack, Text } from '@mantine/core'
+import { omit } from 'remeda'
+import { pathForStudy, pathForStudyDocuments } from '@/lib/paths'
 
-export const StudyProposal: React.FC<{ memberId: string }> = ({ memberId }) => {
+async function uploadFile(file: File, upload: PresignedPost) {
+    const body = new FormData()
+    for (const [key, value] of Object.entries(upload.fields)) {
+        body.append(key, value)
+    }
+    body.append('file', file)
+
+    const response = await fetch(upload.url, {
+        method: 'POST',
+        body,
+    })
+
+    if (!response.ok) {
+        notifications.show({
+            color: 'red',
+            title: 'Failed to upload file',
+            message: await response.text(),
+        })
+    }
+
+    return response.ok
+}
+
+async function uploadCodeFiles(files: File[], upload: PresignedPost, studyJobId: string) {
+    const manifest = new CodeReviewManifest(studyJobId, 'r')
+    for (const codeFile of files) {
+        manifest.files.push(codeFile)
+        await uploadFile(codeFile, upload)
+    }
+
+    const manifestFile = new File([manifest.asJSON], 'manifest.json', { type: 'application/json' })
+    return await uploadFile(manifestFile, upload)
+}
+
+export const StudyProposal: React.FC<{ memberSlug: string }> = ({ memberSlug }) => {
     const router = useRouter()
 
-    const { mutate: createStudy } = useMutation({
+    const studyProposalForm = useForm<StudyProposalFormValues>({
+        mode: 'uncontrolled',
+        validate: zodResolver(studyProposalFormSchema),
+        initialValues: {
+            title: '',
+            piName: '',
+            irbDocument: null,
+            descriptionDocument: null,
+            agreementDocument: null,
+            codeFiles: [],
+        },
+    })
+
+    const { isPending, mutate: createStudy } = useMutation({
         mutationFn: async (formValues: StudyProposalFormValues) => {
-            return await onCreateStudyAction({ memberId, studyInfo: formValues })
+            // Don't send any actual files to the server action, because they can't handle the file sizes
+            const valuesWithoutFiles = omit(formValues, [
+                'agreementDocument',
+                'descriptionDocument',
+                'irbDocument',
+                'codeFiles',
+            ])
+
+            const valuesWithFilenames = {
+                ...valuesWithoutFiles,
+                descriptionDocPath: formValues.descriptionDocument?.name || '',
+                agreementDocPath: formValues.agreementDocument?.name || '',
+                irbDocPath: formValues.irbDocument?.name || '',
+            }
+
+            const { studyId, studyJobId } = await onCreateStudyAction({
+                memberSlug,
+                studyInfo: valuesWithFilenames,
+            })
+
+            if (formValues.irbDocument) {
+                const path = pathForStudyDocuments({ studyId, memberSlug }, StudyDocumentType.IRB)
+                const upload = await signedUrlForStudyFileUploadAction(path)
```

Comment:
I thought we were going to return the url in the study save action and re-use it?  We shouldn't call multiple actions one after another because they are all fetch calls over the network.